openapi/node_test.go

```diff
@@ -0,0 +1,31 @@
+package openapi
+
+import "testing"
+
+func TestOpenAPI_Dig(t *testing.T) {
+	openapi, _ := LoadOpenAPI(absOpenAPI)
+	endpoint := Dig(openapi, testEndpoint)
+	if _, ok := endpoint.(*PathItem); !ok {
```

Comment:
Non blocking: These tests would be more useful if they checked for identity, not just type.

Reply:
Please elaborate? I'm dropping a todo here in the meantime

Reply:
Having a PathItem does not mean having the right PathItem.
Its not too farfetched for a future bug in something like Dig to cause it to always return the first PathItem or for a dev to forget that they should be working with pointers and introduce a pass by value instead.

Not a biggie in this case, which is why its non blocking. I'm resolving the thread